src/plugins/dashboard/public/application/utils/use/use_dashboard_container.tsx

```diff
@@ -126,44 +132,66 @@ const createDashboardEmbeddable = (
 
   const getShouldShowEditHelp = () => {
     return (
-      !savedDash.panels.length &&
-      savedDash.viewMode === ViewMode.EDIT &&
+      !appState.getState().panels.length &&
+      appState.getState().viewMode === ViewMode.EDIT &&
       !dashboardConfig.getHideWriteControls()
     );
   };
 
   const getShouldShowViewHelp = () => {
     return (
-      !savedDash.panels.length &&
-      savedDash.viewMode === ViewMode.VIEW &&
+      !appState.getState().panels.length &&
+      appState.getState().viewMode === ViewMode.VIEW &&
       !dashboardConfig.getHideWriteControls()
     );
   };
 
   const shouldShowUnauthorizedEmptyState = () => {
     const readonlyMode =
-      !savedDash.panels.length &&
+      !appState.getState().panels.length &&
       !getShouldShowEditHelp() &&
       !getShouldShowViewHelp() &&
       dashboardConfig.getHideWriteControls();
     const userHasNoPermissions =
-      !savedDash.panels.length && !visualizeCapabilities.save && !mapsCapabilities.save;
+      !appState.getState().panels.length && !visualizeCapabilities.save && !mapsCapabilities.save;
     return readonlyMode || userHasNoPermissions;
   };
 
   const getEmptyScreenProps = (
     shouldShowEditHelp: boolean,
-    isEmptyInReadOnlyMode: boolean
+    isEmptyInReadOnlyMode: boolean,
+    dashboardContainer: DashboardContainer,
```

Comment:
What's with the lint error pm line 163?
```
'dashboardContainer' is already declared in the upper scope
```

Reply:
+1 probably just need to declare this function outside of the scope of the variable already declared or rename

Reply:
yeah based on https://github.com/opensearch-project/OpenSearch-Dashboards/pull/4346#discussion_r1237748088 the final PR will polish it and provide better structure and readability

Reply:
Take a closer look and here it actually does not need dashboardContainer to be passed into the function since it can just use the dashboardContainer variable.